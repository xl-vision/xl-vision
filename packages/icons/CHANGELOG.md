# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.1-alpha.0](https://github.com/xl-vision/xl-vision/compare/v0.1.0...v0.1.1-alpha.0) (2021-07-11)


### Features

* **Icon:** update icon svgs ([d073aab](https://github.com/xl-vision/xl-vision/commit/d073aab9b959de4956f5712d9b4da4b08c5774e9))






# 0.1.0 (2021-03-21)


### Bug Fixes

* remove same name icon ([28e3620](https://github.com/xl-vision/xl-vision/commit/28e36204e012886f443e5f3a94665eff6e31770c))


### Features

* **Icon:** refactor Icon ([6623ffe](https://github.com/xl-vision/xl-vision/commit/6623ffe8b039fcf514c2381adedc8e4b1529332d))
* **Icons:** support icons ([#2](https://github.com/xl-vision/xl-vision/issues/2)) ([1fb9b5c](https://github.com/xl-vision/xl-vision/commit/1fb9b5ca87f8ba18c931bc1cc14cec3885ece3e5))
