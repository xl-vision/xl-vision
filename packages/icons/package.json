{
  "name": "@xl-vision/icons",
  "version": "0.1.1-alpha.16",
  "description": "xl-vision react components for icon.",
  "author": "Rhys Xia <xrs4433@outlook.com>",
  "license": "MIT",
  "main": "legacy/index.js",
  "module": "modern/index.js",
  "types": "legacy/index.d.ts",
  "unpkg": "dist/index.production.min.js",
  "files": [
    "legacy",
    "modern",
    "dist"
  ],
  "keywords": [
    "xl-vision",
    "react",
    "styles"
  ],
  "homepage": "https://xl-vision.8910.xyz/components/Icon",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/xl-vision/xl-vision.git",
    "directory": "packages/icons"
  },
  "scripts": {
    "build:legacy": "node ../../scripts/compile legacy",
    "build:modern": "node ../../scripts/compile modern",
    "build:dist": "node ../../scripts/build",
    "build": "npm run build:legacy && npm run build:modern && npm run build:dist",
    "generate": "node ./scripts/generate.js",
    "prepublishOnly": "npm run build"
  },
  "bugs": {
    "url": "https://github.com/xl-vision/xl-vision/issues"
  },
  "dependencies": {
    "@xl-vision/styled-engine": "workspace:^",
    "@xl-vision/utils": "workspace:^",
    "clsx": "^1.1.1",
    "prop-types": "^15.8.1"
  },
  "devDependencies": {
    "fs-extra": "^10.0.1",
    "mustache": "^4.2.0",
    "svgo": "^2.8.0"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.15.4",
    "@types/react": ">=16.8.0",
    "react": ">=16.8.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  },
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  }
}
