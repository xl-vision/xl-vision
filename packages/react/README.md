# `@xl-vision/react`

> TODO: description

