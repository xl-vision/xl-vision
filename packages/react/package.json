{
  "name": "@xl-vision/react",
  "version": "0.0.0",
  "description": "xl-vision react components.",
  "author": "Rhys Xia <xrs4433@outlook.com>",
  "license": "MIT",
  "main": "legacy/index.js",
  "module": "modern/index.js",
  "types": "legacy/index.d.ts",
  "files": [
    "legacy",
    "modern"
  ],
  "keywords": [
    "xl-vision",
    "react",
    "styles"
  ],
  "homepage": "https://github.com/xl-vision/xl-vision-next/tree/master/packages/react#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/xl-vision/xl-vision-next.git"
  },
  "scripts": {
    "build:legacy": "node ../../scripts/build legacy",
    "build:modern": "node ../../scripts/build modern",
    "build": "npm run build:legacy && npm run build:modern"
  },
  "bugs": {
    "url": "https://github.com/xl-vision/xl-vision-next/issues"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@types/react": "^16.8.0",
    "@types/react-dom": "^16.8.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "peerDependencies": {
    "@types/react": "^16.8.0 || ^17.0.0",
    "@xl-vision/styled-engine": "^0.0.0",
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  },
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  }
}
