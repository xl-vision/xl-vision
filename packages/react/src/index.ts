export { styled, createGlobalStyles } from './styles';
export { default as ThemeProvider } from './ThemeProvider';
export { default as Transition } from './Transition';
export { default as CSSTransition } from './CSSTransition';
export { default as CollapseTransition } from './CollapseTransition';
export { default as TransitionGroup } from './TransitionGroup';
