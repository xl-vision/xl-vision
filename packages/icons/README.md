# `@xl-vision/icons`

> TODO: description

## Usage

```
import { AccountBookFilled } from '@xl-vision/icons';

```