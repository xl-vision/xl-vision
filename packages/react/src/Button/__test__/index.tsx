import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { CloseCircleFilled } from '@xl-vision/icons';
import { Button, SizeVariant, ThemeProvider } from '@xl-vision/react';

const CloseWrapper = <CloseCircleFilled />;

describe('Button', () => {
  it('Test basic render', () => {
    const { container } = render(
      <div>
        <Button>button</Button>
        <Button color='error'>button</Button>
        <Button variant='outlined'>button</Button>
        <Button prefixIcon={CloseWrapper}>button</Button>
        <Button suffixIcon={CloseWrapper}>button</Button>
        <Button prefixIcon={CloseWrapper} />
        <Button suffixIcon={CloseWrapper} />
        <Button long={true}>button</Button>
        <Button loading={true}>button</Button>
        <Button loading={true} prefixIcon={CloseWrapper} />
        <Button loading={true} prefixIcon={CloseWrapper}>
          button
        </Button>
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Test click', async () => {
    const handleClick = jest.fn();
    const { container, rerender } = render(<Button onClick={handleClick}>button</Button>);

    const el = container.querySelector('button')!;

    const user = userEvent.setup();

    await user.click(el);

    expect(handleClick.mock.calls.length).toBe(1);
    handleClick.mockClear();

    rerender(
      <Button disabled={true} onClick={handleClick}>
        button
      </Button>,
    );

    await user.click(el);

    expect(handleClick.mock.calls.length).toBe(0);
    handleClick.mockClear();

    rerender(
      <Button loading={true} onClick={handleClick}>
        button
      </Button>,
    );

    await user.click(el);

    expect(handleClick.mock.calls.length).toBe(0);
    handleClick.mockClear();
  });

  it('Test component size', () => {
    const componentSizes: Array<SizeVariant> = ['small', 'middle', 'large'];

    const { container, rerender } = render(
      <ThemeProvider>
        <Button>button</Button>
      </ThemeProvider>,
    );

    componentSizes.forEach((componentSize) => {
      rerender(
        <ThemeProvider sizeVariant={componentSize}>
          <Button>button</Button>
        </ThemeProvider>,
      );
      expect(container.querySelector(`.xl-button--size-${componentSize}`)).not.toBe(null);
    });
  });
});

// describe('ButtonGroup', () => {
//   it('基本测试', () => {
//     const wrapper = mount(
//       <Button.Group theme='primary' variant='contained'>
//         <Button>button</Button>
//         <Button theme='error'>button</Button>
//         <Button variant='outlined'>button</Button>
//         <Button prefixIcon={CloseWrapper}>button</Button>
//         <Button suffixIcon={CloseWrapper}>button</Button>
//         <Button prefixIcon={CloseWrapper} />
//         <Button suffixIcon={CloseWrapper} />
//         <Button loading={true}>button</Button>
//         <Button loading={true} prefixIcon={CloseWrapper} />
//         <Button loading={true} prefixIcon={CloseWrapper}>
//           button
//         </Button>
//       </Button.Group>,
//     );

//     expect(wrapper.render()).toMatchSnapshot();

//     wrapper.setProps({
//       size: 'large',
//     });
//     wrapper.update();
//     expect(wrapper.render()).toMatchSnapshot();

//     wrapper.setProps({
//       size: 'small',
//     });
//     wrapper.update();
//     expect(wrapper.render()).toMatchSnapshot();
//   });

//   it('垂直方向', () => {
//     const wrapper = mount(
//       <Button.Group theme='primary' variant='contained' direction='vertical'>
//         <Button>button</Button>
//         <Button theme='error'>button</Button>
//         <Button variant='outlined'>button</Button>
//         <Button prefixIcon={CloseWrapper}>button</Button>
//         <Button suffixIcon={CloseWrapper}>button</Button>
//         <Button prefixIcon={CloseWrapper} />
//         <Button suffixIcon={CloseWrapper} />
//         <Button loading={true}>button</Button>
//         <Button loading={true} prefixIcon={CloseWrapper} />
//         <Button loading={true} prefixIcon={CloseWrapper}>
//           button
//         </Button>
//       </Button.Group>,
//     );

//     expect(wrapper.render()).toMatchSnapshot();

//     wrapper.setProps({
//       size: 'large',
//     });
//     wrapper.update();
//     expect(wrapper.render()).toMatchSnapshot();

//     wrapper.setProps({
//       size: 'small',
//     });
//     wrapper.update();
//     expect(wrapper.render()).toMatchSnapshot();
//   });
// });
