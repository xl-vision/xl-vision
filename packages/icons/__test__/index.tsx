import React from 'react';
import { mount } from 'enzyme';
import { DeleteFilled } from '../src';

describe('Icon', () => {
  it('test render', () => {
    const wrapper = mount(<DeleteFilled />);

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('test ref', () => {
    const fn = jest.fn();
    mount(<DeleteFilled ref={fn} />);

    expect((fn.mock.calls[0] as Array<any>)[0]).toBeInstanceOf(SVGSVGElement);
  });
});
