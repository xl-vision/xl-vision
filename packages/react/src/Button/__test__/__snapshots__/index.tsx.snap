// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test basic render 1`] = `
.c3 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
  overflow: hidden;
  border-radius: inherit;
  pointer-events: none;
}

.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  outline: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0 svg {
  pointer-events: none;
}

.c0 .xl-base-button__ripple {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 0.12;
}

.c0 .xl-base-button__ripple-enter-active {
  -webkit-transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
  transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
}

.c0 .xl-base-button__ripple-leave-active {
  -webkit-transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
  transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
}

.c0 .xl-base-button__ripple-enter-from {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0.1;
}

.c0 .xl-base-button__ripple-leave-to {
  opacity: 0;
}

.c12 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  outline: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: transparent;
  cursor: not-allowed;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.c12::-moz-focus-inner {
  border-style: none;
}

.c12 svg {
  pointer-events: none;
}

.c12 .xl-base-button__ripple {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 0.12;
}

.c12 .xl-base-button__ripple-enter-active {
  -webkit-transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
  transition: all 225ms 0ms cubic-bezier(0,0,0.2,1);
}

.c12 .xl-base-button__ripple-leave-active {
  -webkit-transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
  transition: all 195ms 0ms cubic-bezier(0.4,0,1,1);
}

.c12 .xl-base-button__ripple-enter-from {
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  opacity: 0.1;
}

.c12 .xl-base-button__ripple-leave-to {
  opacity: 0;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 100%;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentcolor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: fill 300ms 0ms cubic-bezier(0.4,0,0.2,1);
}

.c1 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  color: rgba(0,0,0,0.87);
  background-color: #fff;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  padding: 6px 16px;
}

.c1:hover {
  background-color: rgba(247,247,247,1);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c1:focus {
  background-color: rgba(230,230,230,1);
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c4 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  color: #fff;
  background-color: #f44336;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  padding: 6px 16px;
}

.c4:hover {
  background-color: rgba(236,66,54,1);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c4:focus {
  background-color: rgba(220,65,53,1);
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c5 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  background-color: transparent;
  color: rgba(0,0,0,0.87);
  border: 1px solid rgba(0,0,0,0.5);
  padding: 5px 15px;
}

.c5:hover {
  background-color: rgba(0,0,0,0.04);
}

.c5:focus {
  background-color: rgba(0,0,0,0.12);
}

.c9 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 0;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  color: rgba(0,0,0,0.87);
  background-color: #fff;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  padding: 8.2px;
}

.c9:hover {
  background-color: rgba(247,247,247,1);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c9:focus {
  background-color: rgba(230,230,230,1);
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c11 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  width: 100%;
  color: rgba(0,0,0,0.87);
  background-color: #fff;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  padding: 6px 16px;
}

.c11:hover {
  background-color: rgba(247,247,247,1);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c11:focus {
  background-color: rgba(230,230,230,1);
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c13 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  color: rgba(0,0,0,0.87);
  background-color: #fff;
  opacity: 0.38;
  padding: 6px 16px;
}

.c15 {
  -webkit-transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  border-radius: 4px;
  font-size: 0.875rem;
  line-height: 0;
  font-weight: 500;
  -webkit-letter-spacing: 0.089em;
  -moz-letter-spacing: 0.089em;
  -ms-letter-spacing: 0.089em;
  letter-spacing: 0.089em;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-family: 'Roboto','Helvetica Neue','segoe ui','Arial','noto sans',sans-serif;
  color: rgba(0,0,0,0.87);
  background-color: #fff;
  opacity: 0.38;
  padding: 8.2px;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  -webkit-transition: width 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: width 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  padding: 0;
  line-height: 0;
  font-size: 1.4em;
  margin-left: -2px;
  margin-right: 2px;
}

.c6 svg {
  line-height: 1;
}

.c10 {
  display: inline-block;
  vertical-align: middle;
  -webkit-transition: width 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  transition: width 300ms 0ms cubic-bezier(0.4,0,0.2,1);
  padding: 0;
  line-height: 0;
  font-size: 1.4em;
}

.c10 svg {
  line-height: 1;
}

.c8 {
  margin-left: 2px;
  margin-right: -2px;
}

.c14 {
  -webkit-animation: fRjbcp 300ms linear infinite;
  animation: fRjbcp 300ms linear infinite;
}

<div>
  <button
    class="c0 xl-base-button c1 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c4 xl-button xl-button--size-middle xl-button--theme-error xl-button--variant-contained xl-button--elevation"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c5 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-outlined"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c1 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c6 xl-button__prefix"
      >
        <svg
          class="c7 xl-icon"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c1 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="xl-button__inner"
      >
        button
      </span>
      <span
        class="c6 c8 xl-button__suffix"
      >
        <svg
          class="c7 xl-icon"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c9 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--icon"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c10 xl-button__prefix"
      >
        <svg
          class="c7 xl-icon"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c9 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--icon"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c10 xl-button__suffix"
      >
        <svg
          class="c7 xl-icon"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c0 xl-base-button c11 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--long"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c12 xl-base-button xl-base-button--loading c13 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--loading"
    tabindex="-1"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c6 xl-button__prefix xl-button__prefix-appear-from xl-button__prefix-appear-active"
        data-actual-width=""
        data-overflow=""
        data-padding-left=""
        data-padding-right=""
        data-width=""
        style="overflow: hidden; padding-left: 0px; padding-right: 0px;"
      >
        <svg
          class="c7 xl-icon c14 xl-button__prefix--loading"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c12 xl-base-button xl-base-button--loading c15 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--loading xl-button--icon"
    tabindex="-1"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c10 xl-button__prefix"
      >
        <svg
          class="c7 xl-icon c14 xl-button__prefix--loading"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
  <button
    class="c12 xl-base-button xl-base-button--loading c13 xl-button xl-button--size-middle xl-button--theme-default xl-button--variant-contained xl-button--elevation xl-button--loading"
    tabindex="-1"
  >
    <span
      class="c2 xl-base-button__inner"
    >
      <span
        class="c6 xl-button__prefix"
      >
        <svg
          class="c7 xl-icon c14 xl-button__prefix--loading"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
      <span
        class="xl-button__inner"
      >
        button
      </span>
    </span>
    <div
      class="c3 xl-ripple xl-base-button__ripple"
    />
  </button>
</div>
`;
